### DISCUSSION TITLE
How to use selector in Formatter?

### DISCUSSION THREAD

**Discussion by ganesh-sciera** [ghiscoding/Angular-Slickgrid#904] [Q&A] [answered]
prepareGrid() {
    this.columnDefinitions = [
      { id: "albumName", name: "Album Name", field: "albumName", sortable: true, formatter: this.albumNameFormat, width: 300 },
      {
        id: "price",
        name: "Price",
        field: this.country_code.toLowerCase(),
        sortable: true,
        formatter: this.priceFormatter,
        onCellClick: (e: Event, args: OnEventArgs) => {
          this.checkMembership(e, args.dataContext, "price");
          /* setTimeout(() => {
            if(e.target['id'] == 'download') {
              this.downloadAudio(args.dataContext.id)
            } else {
               this.onAddProductToCart(args.dataContext)
            }
          }, 1000); */
        },
      },
     .....
    .....
  ];

  priceFormatter: Formatter = (row, cell, value, columnDef, dataContext) => {
    return "<span><app-albumcart [albumDetailInfo]='" + dataContext + "'></app-albumcart></span>";
  };

But It is not working.. I want to use this selector, which is <app-albumcart> inside the priceFormatter, because, this selector, I have defined the business login and print in the html view... that html view return in this selector...

**Comment by ghiscoding** [comment, marked as answer]
It's all in the [Formatter Wiki](https://github.com/ghiscoding/Angular-Slickgrid/wiki/Formatters#using-angular-component-with-asyncpostrenderer) and related issue #7 

Please ask such questions on Stack Overflow next time and/or search in the Wikis, it has answers for most questions
